interactive_ai/workflows/train/trainer/pyproject.toml

```diff
@@ -44,6 +44,8 @@ build-backend = "hatchling.build"
 [tool.hatch.build.targets.wheel]
 packages = ["trainer"]
 
+[tool.hatch.metadata]
+allow-direct-references = true
```

Comment:
I guess you added this because of the `@` version specifier in otx package.

There is an alternative way to reference a package at a specific revision:
```
dependencies = [
    "otx",
]

[tool.uv.sources]
otx = { git = "https://github.com/open-edge-platform/training_extensions.git", rev = "4107731720f53fc61c51d287a79a23d221c52e84" }
```
Instead of `rev` you can also use `branch = "release/2.5"`. I believe the solution based on `[tool.uv.sources]` is slightly preferable because more flexible and doesn't require direct references. Possibly, it doesn't even require you install git in the Dockerfile, but I didn't verify it.

Reply:
[removed [tool.hatch.metadata], updated installation using branch](https://github.com/open-edge-platform/geti/pull/855/commits/ea2af75cd05b4bb29356bf43f6c2b334a4a6b74c)